### DISCUSSION TITLE
Page Just in dev

### DISCUSSION THREAD

**Discussion by ItsTarik** [vercel/next.js#37076] [Ideas]
### Describe the feature you'd like to request

Hello 👋, just posting this question because i needed to have a page that is available just in development mode, could this be possible or is it supported some how ?

### Describe the solution you'd like

Through next.config maybe ?

### Describe alternatives you've considered

none

**Comment by GabenGar** [comment]
At the start of `getStaticProps()`/`getServerSideProps()`:
```js
const IS_DEVELOPMENT = process.env.NODE_ENV === "development";
if (!IS_DEVELOPMENT) {
    return {
      notFound: true,
    };
}
```

Inside API routes:
```js
const IS_DEVELOPMENT = process.env.NODE_ENV === "development";
if (!IS_DEVELOPMENT) {
    return res.status(404);
}
```

In middleware:
```js
const IS_DEVELOPMENT = process.env.NODE_ENV === "development";
if (!IS_DEVELOPMENT) {
    return NextResponse.redirect("/api/404", 307);
}

```
A bit tricky with middleware considering the latest changes, but just create a "/src/pages/api/404" page which returns 404 response/json and redirect to it.

**Reply by ItsTarik** [reply]
Hello and thanks for this alternative, what i prefer is having Next ignoring this because this kind of pages could affect the build time & bundle size especially with **`SSR`**, anyway it could be a bad idea having such behavior but i found it usefull to have in some contexts like experimenting or having some components preview (without a need for something like storybook)..

**Reply by ItsTarik** [reply]
By the way an api like `page.dev.tsx` or `page._dev.tsx` would be ideal or a custom regex in `next.config` so nextjs internally will ignore while building the app if not in dev mode 🤔
